source/Coop/Mod/Serializers/CustomSerializers/PlayerHeroSerializer.cs

```diff
@@ -159,7 +159,25 @@ public override object Deserialize()
                 .GetField("_exSpouses", BindingFlags.NonPublic | BindingFlags.Instance)
                 .SetValue(hero, lExSpouses);
 
-            return base.Deserialize(hero);
+            
+
+            ConstructorInfo ctor = typeof(HeroDeveloper).GetConstructor(BindingFlags.NonPublic | BindingFlags.Instance,
+                                    null, new Type[] { typeof(Hero) }, null);
+
+            HeroDeveloper newDeveloper = (HeroDeveloper)ctor.Invoke(new object[] { hero });
+            hero.GetType()
+                .GetField("_heroDeveloper", BindingFlags.NonPublic | BindingFlags.Instance)

```

Comment:
Do you happen to know of a way to get this to yield a compile error if the field does not exist? Maybe trough a public property that uses the field internally?

Since they are refactoring game DLLs, these hard coded variable names may compile, but no longer work in a new game version. Some way to at least notice that at compile time would be great.

Reply:
.SetValue will throw a compiler exception due to FieldInfo being null